server/shared/publishing/helpers.js

```diff
@@ -294,6 +303,21 @@ function mapRelationships(relationships, activity) {
   }, {});
 }
 
+async function updateHasChangesFlag(repository, activity) {
```

Comment:
Can you please explain logic behind this ? Consider documenting it.

Reply:
Content in Tailor is published one outline activity at a time which means that activity sidebar badge should indicate whether outline activity is published or not. `activity.modified_at` field tracks the last time any content container or content element within that outline activity is created, updated or deleted.

`has_changes` field is used for repository card badge and it should be `true` if `modified_at > published_at == true` for at least one outline activity within repository. `updateHasChangesFlag` makes sure `has_changes` stays synced after publishing.